### PR TITLE
Tracking Lifetime of Heater Lid Threads [Run Auto Cancelation After 4+ Runs]

### LINKED ISSUE #157: Add temporary heat + thread-lifecycle instrumentation for self-cancel study
## What to build

Temporary, diagnosis-only logging (no control-flow changes) so the lid-heater thread leak and heat behavior are visible directly in the logs — no live shell required. Per `specs/analysis/sentri-self-cancel-study.md` §5.

- A module-level **live lid-worker counter**: each `lid_heater_worker` logs `LID WORKER START tid=… live=N` on entry and `LID WORKER EXIT tid=… live=N` on exit (lock-guarded).
- **ADC read duration** logged per read (to expose I2C stalls > 5 s, the leak's prerequisite).
- **GPIO-21 command** logged (DEBUG) so heater drive density is visible.
- A **run index** + **post-join leak check**: after the lid-thread `join(timeout=5)`, log `LID JOIN DONE run_index=… thread_still_alive=… lid_live=…` — pins a leak to a specific run.

## Triggers addressed

- **Threading / H1 (lid-heater thread leak)** — primary. `live=N` climbing across runs without matching `EXIT` = leak; `thread_still_alive=True` at `LID JOIN DONE` pins the run where the join timed out.
- **Heat / H4** — read-duration + GPIO drive density give visibility into heater behavior across runs.

## Acceptance criteria

- [ ] Each run with a healthy lid thread logs matched `START live=1` → `EXIT live=0`
- [ ] `live=N` reflects only lid workers (no baseline subtraction needed)
- [ ] `tid=` lets a leaked START be matched against a missing EXIT
- [ ] `LID JOIN DONE` logs `thread_still_alive` + `lid_live` immediately after the join
- [ ] No timing change inside the 5 s join window; logging is additive only
- [ ] Changes are clearly marked temporary (revertible per spec §11)

## Blocked by

None - can start immediately.

### PR DESCRIPTION
**Pull request by jackhu23**
## What Changed

Both a spec to diagnose the auto cancelation of a run and implementation of tracking the lifetime of threads have been created.

## Why

https://github.com/AcornGenetics/aquilla-main/issues/157

## Spec

Link to the spec this implements or modifies:
`specs/analysis/sentri-self-cancel-study.md`
`specs/hardware/lid-worker-instrumentation.md`

If no spec exists and this is a non-trivial change, explain why.

## Changes Made

- [x] Source code modified
- [x] Spec updated to match implementation
- [x] New tests written
- [x] Existing tests still pass
- [x] No secrets or `.env` files in this diff
- [ ] ADR written (if an irreversible architectural decision was made)
- [ ] `docs/debugging/known-issues.md` updated (if new edge cases were found)

## Hardware Testing

- [ ] Tested on physical device [Physical testing will be done during issue 162]
- [ ] Tested in simulation mode only — reason: [why physical not possible]
- [ ] Hardware not relevant to this change

## Reviewer Notes

Anything the reviewer should know: tricky parts, things to focus on, known limitations.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "src_basedir": "/opt/aquila"
+  "src_basedir": "/Users/jackh/OneDrive/Documents/acorn-genetics/aquilla-main/"
```
Comment (NicoleCornell8) [inline config.json]: ⚠️ This should remain `/opt/aquila` — the device path. This looks like your local dev path was accidentally committed. Please revert: `git checkout origin/main -- config.json`

**Review by NicoleCornell8 — CHANGES_REQUESTED** [review]
## Changes required before merge

Two issues need to be resolved:

### 1. 🚨 Breaking:  and  have your local dev path hardcoded

Both files were changed from `/opt/aquila` (the device path) to your local OneDrive path. If this merges, `get_src_basedir()` on every deployed Sentri will resolve to a path that doesn't exist on the Pi — this will break data, profiles, history, and the local DB at runtime.

Please revert both files:
```
git checkout origin/main -- config.json config.py
```

### 2. ⚠️ Unrelated: `help.html` rewrite (648 deletions / 1,678 additions)

This is a significant UI change bundled into a PR scoped to lid thread instrumentation. Please either:
- Split it into a separate PR, or
- Update the PR description to explain what changed in Help and why it belongs here

The lid heater instrumentation itself (`regulate.py`, `lid_worker_metrics.py`, `state_run_assay.py`) looks good — the `try/finally` is safer than before and the diagnostics are additive. Happy to approve once the above two items are resolved.

**Comment by NicoleCornell8** [comment]
Following up on the review — just to be explicit on the `help.html` ask: it is a 648-line deletion / 1,678-line addition, which is larger than the actual instrumentation change this PR is for. Even a one-sentence note in the PR description explaining what changed in Help and why it belongs here would be enough. If it is genuinely unrelated work, splitting it keeps the git history clean and makes it easier to bisect if a UI regression shows up later.

File: config.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import os
 from pathlib import Path
 
-DEFAULT_SRC_BASEDIR = "/opt/aquila"
+DEFAULT_SRC_BASEDIR = "/Users/jackh/OneDrive/Documents/acorn-genetics/aquilla-main/"
```
Comment (NicoleCornell8) [inline config.py]: Same here.

**Review by NicoleCornell8 — APPROVED** [review]
done
